src/pass/storage_flatten.cc

```diff
@@ -429,6 +452,30 @@ class StorageFlattener : public IRMutator {
       }
     }
   };
+
+  bool ShapeIsValid(const Array<Expr> &shape) {
+    if (!shape.size())
```

Comment:
it is possible for us to have zero-dimensional tensors(scalar), in which case shape.size() == 0

Reply:
Yes, I believe that it's right way to skip this type of tensors and do not instrument, please fix me if I'm wrong. 
I've added some tests cases. The compute op could be more complicated, at first I was thinking if the store accesses buffer with shape == 0, we should skip annotations and instrumentations, but there are could be example where store accesses zero shape buffer, but load accesses not zero shape.
I've added test cases for this situation. https://github.com/dmlc/tvm/pull/2079/commits/aa0745c08f90268377d9c8785208248281332f64#diff-f904c24c2b3bba4f50a565961755b153R452

Reply:
I see, can you add a comment that zero-dimensional tensor does not need boundary check, so it can give context future readers.

Reply:
Of course, added.